tests/end_to_end/server_test.py

```diff
@@ -48,11 +46,7 @@ def run(self, *args, **kwargs):
             self.otherTeardown()
 
     def runClientCmd(self, client, command):
-        try:
-            client.runCommand(command)
-        except subprocess.CalledProcessError as error:
-            self.server.printDebugInfo()
-            raise error
+        client.runCommand(command)
```

Comment:
It only prints anything if it fails, which is the only time that we would want it to print anything.  This seems correct to me.  What we're essentially choosing between here is, on failure, whether the test user gets told 1) something went wrong and 2) something went wrong and here's information that will allow you to fix it.  2 seems vastly superior to me.  My comment applies for everywhere else you have made a similar change.

In any event, running the test with nose captures the output into a nicely formatted display.